cleanlab/datalab/issue_manager/label.py

```diff
@@ -118,9 +121,19 @@ def _reset(self) -> None:
 
     def find_issues(
         self,
-        pred_probs: np.ndarray,
+        pred_probs: Optional[npt.NDArray] = None,
+        features: Optional[npt.NDArray] = None,
         **kwargs,
     ) -> None:
+        if pred_probs is None:
+            if features is None:
+                raise ValueError("Either pred_probs or features must be provided to find label issues.")
+            # produce out-of-sample pred_probs from features
+            knn = KNeighborsClassifier(n_neighbors=self._k)
+            pred_probs = cross_val_predict(
+                knn, features, self.datalab.labels, cv=5, method="predict_proba"
+            )
```

Comment:
For efficiency and better accuracy, let's not use cross-validation here. You can instead do:

```
knn.fit(features, self.datalab.labels)
pred_probs = knn.predict_proba(features)
```

Then finally you can adjust these `pred_probs` to account for the fact that the datapoint itself was included amongst its neighbors by doing this adjustment for each example `i`:
```
pred_probs[i] = (pred_probs[i] - 1/self._k * onehotlabel[i]) * self._k/(self._k-1)
```
where `onehotlabel` = zeros vector except for a 1 in the index corresponding to the given label of example `i`. Make sure to test the adjustment (aka my math) by verifying the probs still sum to 1 for each example `i`.

Ideally you should vectorize the above adjustment to `pred_probs` across all examples via numpy. The above code is also 2x slower because I call `fit()` and `predict_proba()` separately even though it is technically not required, but I think we can tolerate this inefficiency for ease of implementation.

Reply:
Thanks for the detailed explanation!